--- cmake/modules/DispatchCompilerWarnings.cmake.orig	2020-05-12 13:13:59.619689872 +0300
+++ cmake/modules/DispatchCompilerWarnings.cmake	2020-05-12 13:13:35.216171428 +0300
@@ -6,7 +6,6 @@
   # so that we can use __popcnt64
   add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-fms-extensions>)
 else()
-  add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-Werror>)
   add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-Wall>)
   add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-Wextra>)
 
