diff --git CMakeLists.txt CMakeLists.txt
index ca03e57..82e5e59 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -190,6 +190,7 @@ if(__BUILTIN_TRAP)
 endif()
 
 if(NOT CMAKE_SYSTEM_NAME STREQUAL Android)
+find_package(BlocksRuntime)
   find_package(LibRT)
 endif()
 
diff --git src/CMakeLists.txt src/CMakeLists.txt
index 320b6fc..12bf068 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -1,7 +1,4 @@
 
-if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
-  add_subdirectory(BlocksRuntime)
-endif()
 
 add_library(dispatch
   allocator.c
